File: financialManagerAPI/Entities/Requests/UserRequest.cs
using System.ComponentModel.DataAnnotations;

namespace financialManagerAPI.Entities.Requests
{
    public class UserRequest
    {
        [Required]
        [MinLength(8)]
        public string FullName { get; set; } = null!;

        [Required]
        [EmailAddress]
        public string Email { get; set; } = null!;

        public int MonthlyBudget { get; set; }

        [Range(1, 31)]
        public short BudgetUpdateDay { get; set; }
    }
}

File: financialManagerUI/src/app/entities/requests/user.request.ts
export class UserRequest {
  public fullName: string;
  public email: string;
  public monthlyBudget: number;
  public budgetUpdateDay: number;

  constructor(
    fullName: string,
    email: string,
    monthlyBudget: number = 0,
    budgetUpdateDay: number = 1
  ) {
    this.fullName = fullName;
    this.email = email;
    this.monthlyBudget = monthlyBudget;
    this.budgetUpdateDay = budgetUpdateDay;
  }
}
